Invouce amounts rounded to 2 decimals

jeroenmac opened the pull request:
Updated the following: 
1) round tax amount to 2 decimals places
2) round subtotal amount to 2 decimal places
3) round total amount to 2 decimal places

jeroenmac committed:
round total, subtotal and tax to 2 digits

jeroenmac committed:
version update

jeroenmac committed:
version update

jeroenmac committed:
version update

jeroenmac committed:
version update

jeroenmac committed:
version update

zhangshine commented:
@jeroenmac Hi, thanks for your PR.

Could use [Decimal.quantize](http://devdocs.io/python~2.7/library/decimal#decimal.Decimal.quantize) instead of `round` to round these values to 2 decimals? All tests have failed because `round` returns  different results under different Python versions.

Sorry for my bad english.

jeroenmac committed:
changes:
1) decimal.quantize function used instead of round
2) added function to limit number of digits
3) extended constructor of class

jeroenmac committed:
typo

jeroenmac committed:
changed getdecimals function

jeroenmac commented:
Thank you for your feedback.
1. I've updated the function to use Decimal.quantize
2. I've added the precision parameter to the constructor

zhangshine commented:
@jeroenmac Hi, could you update the test cases? just run `python setup test` and fix it.

This repo is integrate with travis ci, when the tests failed it will show a :x: icon on the right of your commit message. You can click the icon to see the detail.

zhangshine commented on pyinvoice/templates.py:
```diff
@@ -15,8 +15,9 @@
+    precision = None
 
-    def __init__(self, invoice_path, pdf_info=None):
+    def __init__(self, invoice_path, pdf_info=None,precision='0.01'):
```
Add a whitepsace between `,` and `precision`, check [PEP8 style guide](https://www.python.org/dev/peps/pep-0008/) please.

zhangshine commented on pyinvoice/templates.py:
```diff
@@ -216,8 +219,9 @@ def _item_data_and_style(self):
         sum_start_x_index = len(item_data_title) - abs(sum_end_x_index)
 
         # ##### Subtotal #####
+        rounditem_subtotal = self.getroundeddecimal(item_subtotal,self.precision)
```
Same as above

zhangshine commented on pyinvoice/templates.py:
```diff
@@ -226,8 +230,9 @@ def _item_data_and_style(self):
         # Tax total
         if self._item_tax_rate is not None:
             tax_total = item_subtotal * (Decimal(str(self._item_tax_rate)) / Decimal('100'))
+            roundtax_total = self.getroundeddecimal(tax_total,self.precision)
```
Same as above

zhangshine commented on pyinvoice/templates.py:
```diff
@@ -237,13 +242,20 @@ def _item_data_and_style(self):
         # Total
         total = item_subtotal + (tax_total if tax_total else Decimal('0'))
-        item_data.append(('Total', '', '', '', total))
+        roundtotal = self.getroundeddecimal(total,self.precision)
```
And here, thanks

jeroenmac committed:
changed getdecimals function

jeroenmac committed:
updated test_templates

jeroenmac committed:
updated test_templates

jeroenmac commented:
Dear, 

I've updated the test cases. 

Kr, 
Jeroen
